pinset.go

```diff
@@ -83,10 +138,119 @@ func (m *MemPinset) Unpin(req *PinRequest) error {
 
 // Pinned gets the pin status of a path
 func (m *MemPinset) Pinned(path string) (pinned bool, err error) {
-	for _, p := range m.Pins {
+	for _, p := range m.pins {
 		if p == path {
 			return true, nil
 		}
 	}
 	return false, nil
 }
+
+// Pins reads from the list present in the pinset
+func (m *MemPinset) Pins(limit, offset int) (pins []string, err error) {
+	for i, p := range m.pins {
+		if i > offset {
```

Comment:
This condition seems wrong, if I give offset=4 then this loop body will process indexes 0, 1, 2, 3, 4 but skip 5, 6,..